internal/store/postgres/approval_repository.go

```diff
@@ -112,3 +115,34 @@ func (r *ApprovalRepository) BulkInsert(approvals []*domain.Approval) error {
 		return nil
 	})
 }
+
+func (r *ApprovalRepository) AddApprover(approver *domain.Approver) error {
+	m := new(model.Approver)
+	if err := m.FromDomain(approver); err != nil {
+		return fmt.Errorf("parsing approver: %w", err)
+	}
+
+	result := r.db.Create(m)
+	if result.Error != nil {
+		return fmt.Errorf("inserting new approver: %w", result.Error)
+	}
+
+	newApprover := m.ToDomain()
+	*approver = *newApprover
```

Comment:
Is this needed? In the service layer, it is not being used either way.

Reply:
yes, that's because our domain representation of approvers ([]string) is different with the model ([]model.Approver). But just like other creation methods, they usually update the passed reference with the inserted ID

Reply:
Yes. But below is the service layer method which calls this repository method. The updated reference is not used at all in this case. 
```go
func (s *Service) AddApprover(_ context.Context, approvalID, email string) error {
	return s.repo.AddApprover(&domain.Approver{
		ApprovalID: approvalID,
		Email:      email,
	})
}
```

Reply:
because when creating this in the repository I was not thinking about how it will be used in the service layer or somewhere else. Was just thinking about the responsibility of the repository itself with the given function signature (pass approver struct by reference)